api/envoy/extensions/wasm/v3/wasm.proto

```diff
@@ -65,12 +64,18 @@ message PluginConfig {
   // Filter/service configuration used to configure or reconfigure a plugin
   // (proxy_on_configuration).
   google.protobuf.Any configuration = 4;
+
+  // If there is a fatal error on the VM (e.g. exception, abort(), on_start or on_configure return false),
```

Comment:
Is the entire VM tainted or just the plugin? It was unexpected for me that all plugins within the VM are knocked out.

Reply:
In general if the VM crashes (e.g. nullptr, abort()) then we cannot call into the VM safely and all plugins associated with that VM are tainted.

The question is whether or not we want to treat on_configure failures for a particular plugin specially.  I would say no, at least for now.   The reason is that it would be more complex to reason about and I don't think the complexity is warranted without a compelling use case.

Reply:
Didn't see Piotr's comments.